### ISSUE TITLE
Suggestion: Improved Science Minigame

### ISSUE THREAD

**Issue by MyrddinE** [daid/EmptyEpsilon#192]
I'm not a fan of the minigame Science needs to play right now. May I suggest a game of Marco Polo / Sonar instead?

Tap / click inside a rectangular scanning grid and a ping spreads out from your tap. When the ping passes over the (hidden) target, the entire ring blinks. You now know where on a circle the target is. A second ping can narrow it down to a point, and the third or fourth tap can win the minigame. This allows for more strategy in the minigame, as well as more interesting adjustments to the skill.

For example, you could make the ping ring invisible, and have the entire rectangle flash when the ping passes over the target. Then you have to judge the ping distance based on timing (hard to do in the thick of battle). You could have the speed of the ping ring be based on distance to target (making it faster to scan closer targets, and tedious to scan very distant ones).

Another possibility is to keep the existing scans, but have the position of the dials mean something. For example, what if one end of the bar was 400, and the other end 800, and the average position of all one-to-three sliders allowed you to give an approximate answer to weapons about what frequency to dial in (on stage one) and what shields to tune (on stage two). Then science is actively calling out vital data as discovered, rather than spending 20 seconds adjusting sliders with no payout til the end.

**Comment by daid** [comment]
Problem is, I'm a huge fan of the current one. Why do you not like it?
It has a clear skill component, as I'm way way quicker with science scanning then anyone else.

**Comment by Wrongtown** [comment]
I agree, I like the current minigame too and I'm pretty sure I've seen praise on bridgesim.net for it. It also ties nicely into the distorted rings surrounding the Science radar.

That said I have no issue with the idea of adding _more_ minigames and randomly selecting from that population, if people are prepared to build them. That way science never quite knows what they're going to be up against. Getting a minigame you particularly like could be kind of like a "Don't worry captain, I recognize the Myraxxian encryption they're using, this will be over in a-DONE!" moment.

I do like the _idea_ of having locked in a single scan giving you some progress, but there's a danger of painting ourselves into a corner. What if the an enemy ship type has no shields? Or a scenario where the enemy ship can only be damaged via torpedo? Keeping it a little more abstract with a few techno-babble labels for inspiration means you don't trip over edge cases every time you implement something new.

**Comment by MyrddinE** [comment]
I'm not a fan because it requires no thought... you are either scrubbing for clarity, or scrubbing for the inflection point. Both are acts that take time to get better at, but are mindless; there is no strategy involved. No [meaningful decisions](https://learn.canvas.net/courses/3/pages/level-6-dot-1-tips-on-decision-making). In fact, the only meaningful decision science has is who to scan between multiple contacts.

In contrast, Engineering is loaded with meaningful decisions. Trade offs with consequence.

Imagine if you made engineering do a time-wasting minigame every time they wanted to send a repairman to a new room. They have already done the meaningful decision... now you are making them do an interrupting, time wasting minigame that's unrelated to the part of the job that is meaningful and interesting.

Science is all about situational awareness and relaying information. The minigame completely blocks that task. So on second thought, it's not the way the minigame works (though I still think the minigame could be better)... it's that there is a minigame at all!

What if Science had a scanning dish that could be aimed, and only ships in its cone could be scanned? Perhaps it allowed limited vision into nebula? Perhaps the vision range is 15km in all directions, but 30 in the arc of the scanning dish? Then science could have a meaningful decision related to their task, about where to focus their resources (the science dish). The dish could take time to rotate. I would then remove the scanning minigame and simply make it take time to complete (during which no other object can be scanned, and the timer only counts down while the target remains in the scanning cone).

With this change the science officer can continue to provide situational awareness at all times and has choices to make relevant to their role.

**Comment by nallath** [comment]
I partially agree with the meaningfull decisions part. Just because the act (mini game) isn't really meaningfull, does not mean that choosing when & where to do it is also meaningless. The choice to scan certain things (or make assumptions about them) is there. 

There is also something that is skill based in the science stuff; The long range scanners. They give signals on the edges that allow a good science officer to detect objects / ships outside of view range. 

I do agree that this should be improved upon. In the ideal case you would have multiple types of detectors, each with means to fool them (Flares for heat based, gravity generators for mass based, etc)

**Comment by Jhovall** [comment]
I created a few 'science' like minigames for empty epsilon. This was for the larp. 
They are in labview however and talk with http requests currently to EE.
I created a shield frequency changer. A 'hacking' minigame (in which you have to align some moving numbers), a money terminal, A jumpgate triangulator and environmental controls.

**Comment by ajford** [comment]
@Jhovall Can you share? Since it's in Labview, can you maybe upload a few video demos? 

I agree that after a few rounds of scrubbing the sliders, it becomes a very repetitive task. It would be interesting to see either a variety of mini-games, or something a bit more like what @MyrddinE suggests with an active decision on what to allocate resources to. 

It does seem like Helm, Weapons, and Engineering all seem to have active decisions. Helm has to choose direction (influencing Beams & Firing), Weapons has to choose torpedoes loaded, shield freq,  and shielding,  and Eng. has to balance power, repair, and cooling.

**Comment by Jhovall** [comment]
I could upload it. Will probably do it when I am home (And have a decent labview version).

**Comment by Fouindor** [comment]
Thanks ! Does it require a recent version of LabVIEW ?

**Comment by RustyPotato** [comment]
Bump to remind @Jhovall to show his LabView works.
